src/pz390.java

```diff
@@ -8233,7 +8256,20 @@ private void ins_B9XX(){
 				psw_cc = get_int_log_sub_cc();
 				break;		
 	       case 0xFD: // "B9FD=MSRKC,54,340", // B9FD rrfa MSRKC R1,R2,R3 RPI 2202
+		          psw_check = false;
 	    	      ins_setup_rrfa();
+				  rlv1 = (long)reg.getInt(rf2 + 4) * (long)reg.getInt(rf3 + 4);
+				  rv1  = (int)rlv1;
+				  reg.putInt(rf1 + 4, rv1);
+				  if (rlv1 != (long)rv1){
+					  psw_cc = psw_cc3;
+				  } else if (rv1 > 0){
+					  psw_cc = psw_cc2;
+				  } else if (rv1 == 0) {
+					  psw_cc = psw_cc0;
+				  } else {
+                      psw_cc = psw_cc1;
+				  }
 	    	     break;		
```

Comment:
See MSGRKC comment about setting the condition code and possible fixed-point overflow.